# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/ObjLiteral.h
# Commit: ca60758cd534
# Full Hash: ca60758cd534c0f3bb04a7acda15eb41a67e7389
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 12: Use TaggedParserAtomIndex in ObjLiteralStencil. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96919
# ==============================================================================

diff -r 8af30c6d9bb9 -r ca60758cd534 js/src/frontend/ObjLiteral.h
--- a/js/src/frontend/ObjLiteral.h	Sat Nov 14 00:03:36 2020 +0000
+++ b/js/src/frontend/ObjLiteral.h	Sat Nov 14 00:03:45 2020 +0000
@@ -144,6 +144,7 @@
 
 namespace frontend {
 struct CompilationAtomCache;
+struct CompilationStencil;
 class StencilXDR;
 }  // namespace frontend
 
@@ -562,8 +563,8 @@
   }
 };
 
-typedef Vector<const frontend::ParserAtom*, 4, js::SystemAllocPolicy>
-    ObjLiteralAtomVector;
+using ObjLiteralAtomVector =
+    Vector<frontend::TaggedParserAtomIndex, 4, js::SystemAllocPolicy>;
 
 JSObject* InterpretObjLiteral(JSContext* cx,
                               frontend::CompilationAtomCache& atomCache,
@@ -594,7 +595,7 @@
                uint32_t* index) {
     *index = atoms_.length();
     atom->markUsedByStencil();
-    if (!atoms_.append(atom)) {
+    if (!atoms_.append(atom->toIndex())) {
       js::ReportOutOfMemory(cx);
       return false;
     }
@@ -606,8 +607,11 @@
 
 #if defined(DEBUG) || defined(JS_JITSPEW)
   void dump();
-  void dump(JSONPrinter& json);
-  void dumpFields(JSONPrinter& json);
+  void dump(JSONPrinter& json,
+            frontend::CompilationStencil* compilationStencil);
+  void dumpFields(JSONPrinter& json,
+                  frontend::CompilationStencil* compilationStencil);
+
 #endif
 };
 